### DISCUSSION TITLE
Unable to install WSL (Windows Subsystem for Linux) in UTM Windows 11

### DISCUSSION THREAD

**Discussion by kolydart** [utmapp/UTM#4652] [UTM Help (macOS)]
I'm trying to install WSL (Windows Subsystem for Linux) in UTM Windows 11. 
I follow the instructions as shown [here](https://learn.microsoft.com/en-us/windows/wsl/install), but when I get to open Ubuntu app, I get the following error:
```
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x80370102
Please enable the Virtual Machine Platform Windows feature and ensure virtualization is enabled in the BIOS.
For information please visit https://aka.ms/enablevirtualization
```
Is there a UTM configuration that can emulate the "Virtual Machine Platform Windows feature"?
If not, I should submit an issue?
thanks

**Comment by qjasn** [comment]
Because UTM doesn't support nested virtualization,so I think UTM will not support it in the future

**Comment by jamesmortensen** [comment]
The bigger question is does qemu or macOS virtualization framework support nested virtualization? UTM can only provide the features exposed by these upstream projects.

It would be nice to have though. :)

**Comment by giper45** [comment]
Hi, I think that, as M1 does not support [nested virtualization](https://news.ycombinator.com/item?id=27591306) it would not possible to use WSL in Win11 (arm-based platforms)

**Comment by ansidev** [comment]
This command should get things to work.
```
wsl --set-default-version 1
```

**Comment by giper45** [comment]
Hi @ansidev  thank you, after a few extra steps it seems to work. I will proceed to provide a guide as possible.

**Reply by KrisSimon** [reply]
Any progress on the guide? Would be very helpful.

**Comment by giper45** [comment]
Hi @KrisSimon I reproduced the steps, and it was indeed really simple...
however, I post a simple tutorial on Medium for references: 
https://medium.com/@giper45/use-windows-subsystem-linux-on-arm-windows-utm-machines-in-apple-m1-0a2f1f2bd487

hope that it can help you

**Reply by NiHaiden** [reply]
Do note that WSL1 is a really janky and not really usable for some tools. I would just install a Linux VM and use that for Linuxy stuff.

**Reply by eccstartup** [reply]
OK, it works. As a VM, it is not important which version of WSL I am using.
